### ISSUE TITLE
prompted for credentials twice

### ISSUE THREAD

**Issue by allisonmuise** [Esri/crowdsource-reporter#87]
with public content, public maps, public groups and a private or org shared app, I am promted to enter my credentials twice.

This has been reproduced with multiple apps on multiple computers on all desktop browsers (mobile not tested)

Please confirm that you can reproduce this behavior and investigate a fix.

**Comment by CTLocalGovTeam** [comment]
@allisonmuise 

We see that if app is private or org shared, the AGOL sign page is displayed first. After entering credentials the landing page is displayed. If you click the AGOL sign button on landing page, the app logs you in automatically without asking for credentials. Are you seeing the identity manager dialog here?

**Comment by allisonmuise** [comment]
I have posted a video of this behavior to Box and will email credentials for access the data shown in the video

**Comment by CTLocalGovTeam** [comment]
@allisonmuise We were able to repro this. Thank you for sharing the additional data for reproducing this.

We narrowed this down to a trailing slash in sharinghost parameter (after arcgis.com) in default.js

```Javascript
"sharinghost": location.protocol + "//" + "arcgis.com/",
```

If you remove the trailing slash, it works fine. 

This issue occurs only when
* app is private or org-shared 
* app is published with an externally hosted template that is registered with AGOL or Portal
